.github/workflows/push-mimir-build-image.yml

```
@@ -0,0 +1,108 @@
+name: Build and Push mimir-build-image
+
+# configure trigger by pull request or manually
+on:
+  pull_request:
+    types: [opened, synchronize]
+    paths:
+      - mimir-build-image/Dockerfile  
+
+jobs: 
+  build_and_push:
+    runs-on: ubuntu-latest
+    # If the PR is raised by dependabot, then build the image
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Checkout Pull Request Branch
+        run: gh pr checkout ${{ github.event.pull_request.number }}
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Prepare Variables
+        id: prepare
+        run: |
+          echo "path=mimir-build-image/Dockerfile" >> $GITHUB_OUTPUT
+          main_build_image=$(make print-build-image)
+          main_image_tag=$(echo $main_build_image | cut -d ':' -f 2)
+          image_name=$(echo $main_build_image | cut -d ':' -f 1)
+          echo "image=$image_name" >> $GITHUB_OUTPUT
+          echo "main_image_tag=$main_image_tag" >> $GITHUB_OUTPUT
+
+      - name: Compute Image Tag
+        id: compute_hash
+        run: |
+          current_hash=$(md5sum ${{ steps.prepare.outputs.path }} | awk '{print substr($1, 0, 10)}')
+          echo "the file path is ${{ steps.prepare.outputs.path }}"
+          echo "build tag is $current_hash"
+          tag="pr${{ github.event.pull_request.number }}-$current_hash"
+          echo "tag=$tag" >> $GITHUB_OUTPUT
+
+      - name: Check Should Build Image
+        id: check_build
+        run: |
+          echo "Checking if image should be built"
+          if skopeo inspect --raw "docker://${{ steps.prepare.outputs.image }}:${{ steps.compute_hash.outputs.tag }}" >/dev/null 2>&1; then
+            echo "build=false" >> $GITHUB_OUTPUT
+            echo "Tag ${{ steps.compute_hash.outputs.tag }} exists"
+          else
+            echo "Tag ${{ steps.compute_hash.outputs.tag }} does not exist"
+            echo "build=true" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Add Comment to the PR
+        id: notification
+        run: | 
+          if [ ${{ steps.check_build.outputs.build }} == 'true' ]; then
+           gh pr comment $PR_URL --body "**Building new version of mimir-build image**. After image is built and pushed to docker hub, \
+           new commit will be added to this PR with new image version $IMAGE:$TAG. This might take around 1 hour."
+          else
+            echo "This PR will not trigger a build of the mimir-build-image"
+            gh pr comment $PR_URL --body "**Not building new version of mimir-build image, image $IMAGE:$TAG already exits.**"
+          fi
```

[nit] clarify messages

```suggestion
          if [ ${{ steps.check_build.outputs.build }} == 'true' ]; then
           gh pr comment $PR_URL --body "**Building new version of `mimir-build-image`**. After image is built and pushed to Docker Hub, \
           a new commit will automatically be added to this PR with new image version `$IMAGE:$TAG`. This can take up to 1 hour."
          else
            echo "This PR will not trigger a build of mimir-build-image"
            gh pr comment $PR_URL --body "**Not building new version of `mimir-build image`**. This PR modifies `mimir-build-image/Dockerfile`, but the image `$IMAGE:$TAG` already exists."
          fi
```